### PR TITLE
DietPi-Imager | Move from 7z to xz for image compression

### PR DESCRIPTION
**Pull request by MichaIng**

### CODE REVIEW

**Comment by dirkhh** [comment]
Why? It's significantly slower. As in a LOT slower. For fairly marginal gain.
May I lobby you to consider moving to LZ4 instead? https://github.com/lz4/lz4

**Comment by Joulinar** [comment]
using xz you could flash an image directly using the archive to SD card, while it was needed to extract on 7z first.

**Comment by dirkhh** [comment]
> using xz you could flash an image directly using the archive to SD card, while it was needed to extract on 7z first.

So this is a comment about some SD card writing tools like Pi Imager being able to convert `.xz` images "on the fly" when writing an SD card? Well, if that's the goal, then please consider `.zstd`. Pi Imager supports that just as well, and it is once again dramatically faster than `.xz`

Put differently - the only time `.xz` should be the preferred choice is when size matters _at all cost_; whenever you a looking for something that provides a reasonable trade-off between time and compression, there are much better choices.
Just the impact on duration of GitHub Action runs (and, turnaround time when doing test builds) is huge. In a couple of quick tests here, compression took almost as long as the rest of the build using `.xz`.

**Comment by MichaIng** [comment]
7z and xz both use LZMA2 (by default) and hence have similar (de)compression stats. Main reason indeed is the simple single file (de)compression (it automatically adds/removes the xz extension resp. by default removes the archive/original, like gzip, bzip2 and also zstd tools do) and pipe support.

Decompression speed is acceptable and usually not the bottleneck when flashing to a drive directly. I like to keep using LZMA2 as IMO download bandwidth and disk space usage is more a concern then (de)compression speed, and it still achieves a better compression ratio than zstd on highest level (which also reduces its performance benefit).

Basically we align it with Armbian, APT repos, Linux kernel sources etc, which all use xz nowadays, while e.g. zstd is not as widely used/known. A practical example is that 7zip (which I use on Windows as standard archiver) does not support zstd yet, so it is the only compression algorithm I need an extra tool for, annoyingly...

**Comment by dirkhh** [comment]
Thanks for the detailed response.
I understand your reasoning.
